### PR TITLE
Fix/1811  switch app agenda

### PR DESCRIPTION
**Pull request by Skylsmoi**
https://github.com/tracim/tracim/issues/1811

## Checkpoints

- [x] Code is clear enoug
- [x] If there are FIXME in the code, then there are associated issues and issue is referenced in the FIXME
- [x] If there are TODO, NOTE or HACK in code, date and developer initials are present
- [x] Automated tests have been written (covering feature or non regression), or *at least* an issue has been created for test implementation

### CODE REVIEW

File: frontend/src/component/Sidebar/WorkspaceListItem.jsx [hunk 1]
```diff
@@ -64,7 +67,10 @@ const WorkspaceListItem = props => {
       <AnimateHeight duration={500} height={props.isOpenInSidebar ? 'auto' : 0}>
         <ul className='sidebar__content__navigation__workspace__item__submenu'>
           {props.allowedAppList.map(aa =>
-            <li key={aa.slug}>
+            <li
+              data-cy={`sidebarSubdropdown-${aa.slug}`}
```
Comment (basilelegal) [inline frontend/src/component/Sidebar/WorkspaceListItem.jsx]: we used _ as a separator on every other data-cy, we should stick to it

File: functionnal_tests/cypress/integration/app_agenda/switching_app_agenda.jsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        .contains(workspace1.label)
+
+      cy.get('#agendaIframe')
+        .invoke('attr', 'data-config')
+        .should('contain', `agenda/workspace/${workspace1.workspace_id}/`)
+
+      cy.get(`.sidebar__content__navigation__workspace__item__name[title="${workspace2.label}"]`)
+        .click()
+
+      cy.get(`[data-cy="sidebar__content__navigation__workspace__item_${workspace2.workspace_id}"]`)
```
Comment (basilelegal) [inline functionnal_tests/cypress/integration/app_agenda/switching_app_agenda.jsx]: getTag(selectorName:SELECTORS.WORKSPACE_MENU, params:{workspaceId: workspace2.workspace_id}} does the same

File: frontend/src/component/Sidebar/WorkspaceListItem.jsx (hunk 1 above)
Comment (lebouquetin) [reply frontend/src/component/Sidebar/WorkspaceListItem.jsx]: what means `aa` in `${aa.slug}`. This is a bad variable name. allowedApp (if it is this) must be used in this type of case. Please avoid to save chars this way.

File: functionnal_tests/cypress/integration/app_agenda/switching_app_agenda.jsx (hunk 2 above)
Comment (Skylsmoi) [reply functionnal_tests/cypress/integration/app_agenda/switching_app_agenda.jsx]: done

File: frontend/src/component/Sidebar/WorkspaceListItem.jsx (hunk 1 above)
Comment (Skylsmoi) [reply frontend/src/component/Sidebar/WorkspaceListItem.jsx]: done (both)

**Pull request merged**